vm/actor/src/builtin/puppet/mod.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+
+use crate::check_empty_params;
+use address::Address;
+use encoding::{tuple::*, Cbor};
+use ipld_blockstore::BlockStore;
+use num_bigint::bigint_ser;
+use runtime::{ActorCode, Runtime};
+use serde::{Deserialize, Serialize};
+use vm::{ActorError, ExitCode, MethodNum, Serialized, TokenAmount, METHOD_CONSTRUCTOR};
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    SendMethod = 2,
+    SendMarshalCBORFailure = 3,
+    ReturnMarshalCBORFailure = 4,
+    RuntimeTransactionMarshalCBORFailure = 5,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SendParams {
+    pub to: Address,
+    #[serde(with = "bigint_ser")]
+    pub value: TokenAmount,
+    pub method: u64,
+    pub params: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SendReturn {
+    pub return_bytes: Serialized,
+    pub code: ExitCode,
+}
+
+#[derive(Default, Serialize, Deserialize)]
+pub struct FailToMarshalCBOR {}
```

Comment:
This doesn't fail?

Reply:
Nope, serde really is magic

Reply:
No, I mean the intention of this struct is that it _should_ fail

Reply:
> No, I mean the intention of this struct is that it _should_ fail

Yes my bad completely skipped over the implementation for the Cbor trait